src/mono/cmake/configure.cmake

```diff
@@ -163,4 +163,8 @@ if(HOST_IOS)
     set(HAVE_EXECVP 0)
     set(HAVE_SIGNAL 0)
   endif()
-endif()
\ No newline at end of file
+endif()
+
+if(MAC_CATALYST)
```

Comment:
this should be `HOST_MACCATALYST` to be consistent with all the other platforms, you'll need to add a section for it in https://github.com/dotnet/runtime/blob/master/src/mono/CMakeLists.txt#L186 (see my other comment)